assets/javascripts/discourse/components/fields/da-period-field.gjs

```diff
@@ -17,15 +17,8 @@ export default class PeriodField extends BaseField {
   constructor() {
     super(...arguments);
 
-    if (!this.args.field.metadata.value) {
-      this.args.field.metadata.value = new TrackedObject({
-        interval: 1,
-        frequency: null,
-      });
-    }
-
-    this.interval = this.args.field.metadata.value.interval;
-    this.frequency = this.args.field.metadata.value.frequency;
+    this.interval = this.args.field.metadata.value?.interval || 1;
```

Comment:
What was the problem with this code?

Reply:
Ah I see, generally you can get around this, by wrapping in a `next(() => {})` block, this might be better than the `ensureValue()` function